lib/vsc/administration/ldapsync.py

```diff
@@ -121,7 +121,16 @@ def sync_altered_accounts(self, last, dry_run=True):
                 logging.warning("Converting unicode to ascii for gecos resulting in %s", gecos)
             logging.debug('fetching public key')
 
-            public_keys = [str(x.pubkey) for x in self.client.get_public_keys(account.vsc_id)]
+            public_keys = []
+            for something in self.client.get_public_keys(account.vsc_id):
+                logging.debug("Got pubkey %s for account %s", something.pubkey, account.vsc_id)
+                try:
+                    pk = str(something.pubkey)
+                except UnicodeEncodeError:
+                    pk = something.pubkey.encode('ascii', 'ignore')
+                    logging.warning("Converting unicode to ascii for pubkey resulting in %s", pk)
```

Comment:
is there any value in trying to log the original key? because we are modifying the key
@itkovian but i'm curious what triggered this. isn't the `.pubkey` the actual key, so what unicode chars could be in it or are even allowed?

Reply:
There were unicode chars in the thingie after the key :)